pkg/asset/installconfig/ibmcloud/client.go

```diff
@@ -513,7 +513,7 @@ func (c *Client) GetCOSInstanceByName(ctx context.Context, cosName string) (*res
 		}
 	}
 
-	return nil, fmt.Errorf("failed to find cos instance %s", cosName)
+	return nil, nil
```

Comment:
What about creating a `cosNotFoundError` and returning that instead of `nil` or a dynamic error? That way you can check in the callee:
```golang
    cosInstance, err := client.GetCOSInstanceByName(ctx, cosInstanceName)
	if err != nil {
        if !errors.Is(err, costNotFoundError) {
		    return fmt.Errorf("failed checking cos instance for ignition data: %w", err)
        }
		// Attempt to create the COS Instance, since it was not found.
		logrus.Debugf("creating cos instance: %s", cosInstanceName)
        [...]
    }
```

Reply:
Done